src/main/java/com/github/theholywaffle/teamspeak3/api/wrapper/Client.java

```diff
@@ -28,174 +28,201 @@
-		int[] serverGroupIds = getServerGroups();
-		for (int s : serverGroupIds) {
-			if (s == serverGroupId) return true;
-		}
-		return false;
-	}
-
-	/**
-	 * Utility method that does a linear search on the array of server group IDs returned
-	 * by {@link #getServerGroups()} and returns {@code true} if that array contains
-	 * the ID of the given server group.
-	 *
-	 * @param serverGroup
-	 * 		the server group to search for
-	 *
-	 * @return whether this client is a member of the given server group
-	 */
-	public boolean isInServerGroup(ServerGroup serverGroup) {
-		return isInServerGroup(serverGroup.getId());
-	}
-
-	public boolean isInputHardware() {
-		return getBoolean(ClientProperty.CLIENT_INPUT_HARDWARE);
-	}
-
-	public boolean isInputMuted() {
-		return getBoolean(ClientProperty.CLIENT_INPUT_MUTED);
-	}
-
-	public boolean isOutputHardware() {
-		return getBoolean(ClientProperty.CLIENT_OUTPUT_HARDWARE);
-	}
-
-	public boolean isOutputMuted() {
-		return getBoolean(ClientProperty.CLIENT_OUTPUT_MUTED);
-	}
-
-	public boolean isPrioritySpeaker() {
-		return getBoolean(ClientProperty.CLIENT_IS_PRIORITY_SPEAKER);
-	}
-
-	public boolean isRecording() {
-		return getBoolean(ClientProperty.CLIENT_IS_RECORDING);
-	}
-
-	public boolean isRegularClient() {
-		return getType() == 0;
-	}
-
-	public boolean isServerQueryClient() {
-		return getType() == 1;
-	}
-
-	public boolean isTalking() {
-		return getBoolean("client_flag_talking");
-	}
+    public Client(Map<String, String> map) {
+        super(map);
+    }
+
+    public boolean canTalk() {
+        return getBoolean(ClientProperty.CLIENT_IS_TALKER);
+    }
+
+    public String getAwayMessage() {
+        return get(ClientProperty.CLIENT_AWAY_MESSAGE);
+    }
+
+    public int getChannelGroupId() {
+        return getInt(ClientProperty.CLIENT_CHANNEL_GROUP_ID);
+    }
+
+    public int getChannelId() {
+        return getInt(ClientProperty.CID);
+    }
+
+    /**
+     * Utility method. This method will return a client uri.
+     * A client uri can be used to reference a client.
+     * Example: [URL=client://5/thisisanuid=~Jesus]Jesus[/URL]
+     *
+     * @return clientUri
+     */
+    public String getClientUri() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("[URL=client://");
+        sb.append(getId());
+        sb.append("/");
+        sb.append(getUniqueIdentifier());
+        sb.append("~");
+        String encodedNick;
+        try {
+            //We will encode the nickname, because there are people, that are using incompatible chars in names.
+            encodedNick = URLEncoder.encode(getNickname(), "UTF-8"); //"The World Wide Web Consortium Recommendation states that UTF-8 should be used. Not doing so may introduce incompatibilites." Source: https://docs.oracle.com/javase/7/docs/api/java/net/URLEncoder.html#encode(java.lang.String,%20java.lang.String)
+        } catch (UnsupportedEncodingException e) {
+            encodedNick = getNickname();
```

Comment:
Just re-throw the exception as a `new RuntimeException(e)`. UTF-8 is one of the charsets that every implementation of Java must support. Returning something that's probably-most-of-the-time-correct is not an option in this case.